### ISSUE TITLE
Apps using X freezes and new ones does not start

### ISSUE THREAD

**Issue by Ueland** [microsoft/wslg#450]
I am testing an environment running on Docker, that is located within WSL2 and utilizing WSLG. I added Pulseaudio in the Docker container in order to get sound from the Docker environment into Windows. It did work but i also notice that all X apps froze, and starting up new ones causes the process starting them to hang. So I suspect a correlation ot PulseAudio here.

# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.22000.184]
Your Distribution version: Ubuntu 20.04
Your WSLg version: [Open 'Apps and Features' and check the version of 'Windows Subsystem for Linux GUI app support', e.g: 0.2.3.13]
```

**WSL logs**: 

[stderr.log](https://github.com/microsoft/wslg/files/7147628/stderr.log)
[pulseaudio.log](https://github.com/microsoft/wslg/files/7147626/pulseaudio.log)
[weston.log](https://github.com/microsoft/wslg/files/7147627/weston.log)

**Comment by hello123568** [comment]
too hard

**Comment by hideyukn88** [comment]
@Ueland, thanks for reporting the issue. Just for clarification, you don't experience the hang if no pulseaudio is enabled from docker container? Does hang happen only after starting app which uses audio? or any kind of app? thanks!

**Comment by Ueland** [comment]
Correct, it was fine until that. When it triggered, i could not start any new app that uses UI/X.

**Comment by hideyukn88** [comment]
@Ueland, thanks. While I can't see any relationship with pulseaudio, but I see below error logged in your weston.log, this ends up no window visible on desktop, so it can be appeared as hang. This is reported at https://github.com/microsoft/wslg/issues/179. Would you please check whenever you encounter this issue, does below error get logged always? thanks!

`[12:30:22.458] rdp_allocate_shared_memory: Failed to open "/mnt/shared_memory/{78873564-aa69-4224-b3a7-11619533f43c}" with error: Input/output error`

**Comment by ThaDaVos** [comment]
I am having the same issue but then with a simple application as Dbeaver-CE (which as far as I know doesn't use audio) - after it freezes, I cannot start any new applications

**Comment by Ueland** [comment]
@hideyukn88, yes, I will report back if I manage to trigger the issue again.
